### ISSUE TITLE
Installer script stops after installing php5-sqlite

### ISSUE THREAD

**Issue by allo-** [pi-hole/pi-hole#2771]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

3 (I looked shortly into the install script and know a bit of bash programming, but did not patch anything or similar steps)

---
**Expected behaviour:**

The installer should run without problems

**Actual behaviour:**

The installer stops after installing php-sqlite3

**Steps to reproduce:**

Run the installer in a minimal debian environment

**Troubleshooting undertaken, and/or other relevant information:**

``apt install bc``

Afterwards the script continued after the php-sqlite3 step.

**Comment by DL6ER** [comment]
I tried with Debian 9.7 (DigitalOcean) and can install `php-sqlite3` without having `bc`:
```
root@debian-s-1vcpu-1gb-fra1-01:~# apt remove bc
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package 'bc' is not installed, so not removed
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
```
```
root@debian-s-1vcpu-1gb-fra1-01:~# apt install php-sqlite3
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  php-common php7.0-common php7.0-sqlite3 psmisc
The following NEW packages will be installed:
  php-common php-sqlite3 php7.0-common php7.0-sqlite3 psmisc
0 upgraded, 5 newly installed, 0 to remove and 1 not upgraded.
Need to get 1.049 kB of archives.
After this operation, 5.524 kB of additional disk space will be used.
Do you want to continue? [Y/n] 
Get:1 http://mirrors.digitalocean.com/debian stretch/main amd64 psmisc amd64 22.21-2.1+b2 [123 kB]
Get:2 http://mirrors.digitalocean.com/debian stretch/main amd64 php-common all 1:49 [14,1 kB]
Get:3 http://mirrors.digitalocean.com/debian stretch/main amd64 php7.0-common amd64 7.0.33-0+deb9u3 [882 kB]
Get:4 http://mirrors.digitalocean.com/debian stretch/main amd64 php7.0-sqlite3 amd64 7.0.33-0+deb9u3 [24,9 kB]
Get:5 http://mirrors.digitalocean.com/debian stretch/main amd64 php-sqlite3 all 1:7.0+49 [5.052 B]
Fetched 1.049 kB in 0s (5.265 kB/s)
Selecting previously unselected package psmisc.
(Reading database ... 29465 files and directories currently installed.)
Preparing to unpack .../psmisc_22.21-2.1+b2_amd64.deb ...
Unpacking psmisc (22.21-2.1+b2) ...
Selecting previously unselected package php-common.
Preparing to unpack .../php-common_1%3a49_all.deb ...
Unpacking php-common (1:49) ...
Selecting previously unselected package php7.0-common.
Preparing to unpack .../php7.0-common_7.0.33-0+deb9u3_amd64.deb ...
Unpacking php7.0-common (7.0.33-0+deb9u3) ...
Selecting previously unselected package php7.0-sqlite3.
Preparing to unpack .../php7.0-sqlite3_7.0.33-0+deb9u3_amd64.deb ...
Unpacking php7.0-sqlite3 (7.0.33-0+deb9u3) ...
Selecting previously unselected package php-sqlite3.
Preparing to unpack .../php-sqlite3_1%3a7.0+49_all.deb ...
Unpacking php-sqlite3 (1:7.0+49) ...
Setting up psmisc (22.21-2.1+b2) ...
Setting up php-common (1:49) ...
Created symlink /etc/systemd/system/timers.target.wants/phpsessionclean.timer → /lib/systemd/system/phpsessionclean.timer.
Setting up php7.0-common (7.0.33-0+deb9u3) ...

Creating config file /etc/php/7.0/mods-available/calendar.ini with new version

Creating config file /etc/php/7.0/mods-available/ctype.ini with new version

Creating config file /etc/php/7.0/mods-available/exif.ini with new version

Creating config file /etc/php/7.0/mods-available/fileinfo.ini with new version

Creating config file /etc/php/7.0/mods-available/ftp.ini with new version

Creating config file /etc/php/7.0/mods-available/gettext.ini with new version

Creating config file /etc/php/7.0/mods-available/iconv.ini with new version

Creating config file /etc/php/7.0/mods-available/pdo.ini with new version

Creating config file /etc/php/7.0/mods-available/phar.ini with new version

Creating config file /etc/php/7.0/mods-available/posix.ini with new version

Creating config file /etc/php/7.0/mods-available/shmop.ini with new version

Creating config file /etc/php/7.0/mods-available/sockets.ini with new version

Creating config file /etc/php/7.0/mods-available/sysvmsg.ini with new version

Creating config file /etc/php/7.0/mods-available/sysvsem.ini with new version

Creating config file /etc/php/7.0/mods-available/sysvshm.ini with new version

Creating config file /etc/php/7.0/mods-available/tokenizer.ini with new version
Setting up php7.0-sqlite3 (7.0.33-0+deb9u3) ...

Creating config file /etc/php/7.0/mods-available/sqlite3.ini with new version

Creating config file /etc/php/7.0/mods-available/pdo_sqlite.ini with new version
Setting up php-sqlite3 (1:7.0+49) ...
```

**Comment by DL6ER** [comment]
@allo- Do you have a particular reason for installing `bc` or would it maybe have continued after having installed *any* package manually?

**Comment by allo-** [comment]
php-sqlite3 is the last command that works correctly, afterwards the script stopped without any further error message. I googled the last line and found some advice to install a few packages with apt-get. The only additional package that was installed by apt-get was ``bc``.

**Comment by DL6ER** [comment]
Hmm, `bc` shouldn't be used by anything. Can you give some more details what you mean by

> a minimal debian environment

so I can try to reproduce this issue?

**Comment by allo-** [comment]
I used a fresh LXC, so this should be quite easy to reproduce, when you have LXC installed:

Build a container using:

    lxc-create lxc-create -t download -n pihole -- -d debian -r stretch -a amd64

You will end up having a minimal container, that for example does not even have wget preinstalled.

Install wget and then run the ``basic-install.sh`` script. For me everything runs up to 

      [i] Checking for php5-sqlite (will be installed)

And then the script terminates without any error message. On the first run, I even thought it may be finished, when it stopped at this point.

**Comment by allo-** [comment]
I tried in a new container. It is *not* bc.
The script terminates here:

```
+ echo -e '\r\033[K  [i] Checking for php-sqlite3 (will be installed)'
  [i] Checking for php-sqlite3 (will be installed)
+ installArray+=("${i}")
+ [[ 3 -gt 0 ]]
+ test_dpkg_lock
+ i=0
+ fuser /var/lib/dpkg/lock
+ return 0
+ debconf-apt-progress -- apt-get --yes --no-install-recommends install resolvconf php-cgi php-sqlite3

```

with status code 100

**Comment by dschaper** [comment]
Try `sudo apt-get --yes --no-install-recommends install resolvconf php-cgi php-sqlite3` and see why it's failing.

**Comment by allo-** [comment]
Running the apt-get command I get

```
# apt-get --yes --no-install-recommends install resolvconf php-cgi php-sqlite3
Reading package lists... Done
Building dependency tree       
Reading state information... Done
php-cgi is already the newest version (1:7.0+49).
php-sqlite3 is already the newest version (1:7.0+49).
resolvconf is already the newest version (1.79).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
11 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up ucf (3.0036) ...
Setting up resolvconf (1.79) ...
Created symlink /etc/systemd/system/sysinit.target.wants/resolvconf.service → /lib/systemd/system/resolvconf.service.
Setting up php7.0-common (7.0.33-0+deb9u3) ...

Creating config file /etc/php/7.0/mods-available/calendar.ini with new version

Creating config file /etc/php/7.0/mods-available/ctype.ini with new version

Creating config file /etc/php/7.0/mods-available/exif.ini with new version

Creating config file /etc/php/7.0/mods-available/fileinfo.ini with new version

Creating config file /etc/php/7.0/mods-available/ftp.ini with new version

Creating config file /etc/php/7.0/mods-available/gettext.ini with new version

Creating config file /etc/php/7.0/mods-available/iconv.ini with new version

Creating config file /etc/php/7.0/mods-available/pdo.ini with new version

Creating config file /etc/php/7.0/mods-available/phar.ini with new version

Creating config file /etc/php/7.0/mods-available/posix.ini with new version

Creating config file /etc/php/7.0/mods-available/shmop.ini with new version

Creating config file /etc/php/7.0/mods-available/sockets.ini with new version

Creating config file /etc/php/7.0/mods-available/sysvmsg.ini with new version

Creating config file /etc/php/7.0/mods-available/sysvsem.ini with new version

Creating config file /etc/php/7.0/mods-available/sysvshm.ini with new version

Creating config file /etc/php/7.0/mods-available/tokenizer.ini with new version
Setting up php7.0-readline (7.0.33-0+deb9u3) ...

Creating config file /etc/php/7.0/mods-available/readline.ini with new version
Setting up php7.0-opcache (7.0.33-0+deb9u3) ...

Creating config file /etc/php/7.0/mods-available/opcache.ini with new version
Setting up php7.0-sqlite3 (7.0.33-0+deb9u3) ...

Creating config file /etc/php/7.0/mods-available/sqlite3.ini with new version

Creating config file /etc/php/7.0/mods-available/pdo_sqlite.ini with new version
Setting up php7.0-json (7.0.33-0+deb9u3) ...

Creating config file /etc/php/7.0/mods-available/json.ini with new version
Setting up php-sqlite3 (1:7.0+49) ...
Setting up php7.0-cli (7.0.33-0+deb9u3) ...
update-alternatives: using /usr/bin/php7.0 to provide /usr/bin/php (php) in auto mode
update-alternatives: using /usr/bin/phar7.0 to provide /usr/bin/phar (phar) in auto mode
update-alternatives: using /usr/bin/phar.phar7.0 to provide /usr/bin/phar.phar (phar.phar) in auto mode

Creating config file /etc/php/7.0/cli/php.ini with new version
Setting up php7.0-cgi (7.0.33-0+deb9u3) ...
update-alternatives: using /usr/bin/php-cgi7.0 to provide /usr/bin/php-cgi (php-cgi) in auto mode
update-alternatives: using /usr/lib/cgi-bin/php7.0 to provide /usr/lib/cgi-bin/php (php-cgi-bin) in auto mode

Creating config file /etc/php/7.0/cgi/php.ini with new version
Setting up php-cgi (1:7.0+49) ...
Processing triggers for resolvconf (1.79) ...
W: APT had planned for dpkg to do more than it reported back (27 vs 34).
   Affected packages: resolvconf:armhf ucf:armhf

```

**Comment by dschaper** [comment]
That command ended in a warning, so that warning was passed up to the installer script. If there are any errors or warnings then it will not install. Running the command manually should process the warnings and then allow the installer to complete.

**Comment by allo-** [comment]
Yes, it seems to work afterwards. But why did it "plan more than it reported back" and what is the bug/issue there?

**Comment by dschaper** [comment]
>"plan more than it reported back"

I'm not sure where this quote comes from?

**Comment by allo-** [comment]
It is in the last line in the apt-get output when running it manually:
```
...
Processing triggers for resolvconf (1.79) ...
W: APT had planned for dpkg to do more than it reported back (27 vs 34).
   Affected packages: resolvconf:armhf ucf:armhf
```

**Comment by dschaper** [comment]
That's on operating system level error and not one that is generated by Pi-hole. You'd have to check on a Debian or Raspbian forum for that information.

**Comment by allo-** [comment]
If you think so I close the bug here, because it works for me now, too.
I think you should consider at least adding some warning "Installation aborted: apt-get failed" or similar, but you need to know how many edge cases you want to handle.

**Comment by seaniedan** [comment]
I was trying to install pi-hole on a raspbian lite image and had this problem. I ran 
sudo apt install bc
then retried the install and it worked.

**Comment by allo-** [comment]
@seaniedan Yes, it is definitively that the base image does not contain ``bc``. The question here is, if the installer-script should install it automatically or see it as bug or OS error (i.e. think images without bc are incomplete as base for pi-hole).

When you install some stuff before pi-hole you probably often install ``bc`` as a dependency of some other package, so the pi-hole installer often will work by chance.

**Comment by Ardakilic** [comment]
I also suffered from the same error just now with Armbian buster with Orange pi zero. Fresh OS installation. It failed because some packages needed configuration.

I tried to install bc, but it said it was already installed, but after that it configured some PHP packages, and after that wen I re-ran the script it installed just fine.

What I believe is that somehow installer doesn't wait for the packages to be configured so it throws error.
